### ISSUE TITLE
The argument of `take()`, `take_along_dim()` and `gather()` better be only one of `index` or `indices` for consistency.

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#130512]
### 🚀 The feature, motivation and pitch

There are the similar functions [take()](https://pytorch.org/docs/stable/generated/torch.take.html), [take_along_dim()](https://pytorch.org/docs/stable/generated/torch.take_along_dim.html) and [gather()](https://pytorch.org/docs/stable/generated/torch.gather.html).

But their aguments are not consistent being used either `index` or `indices` as shown below:

`take()`:

> - index (LongTensor) – the indices into tensor

`take_along_dim()`:

> - indices (tensor) – the indices into input. Must have long dtype.

`gather()`:

> - index (LongTensor) – the indices of elements to gather

### Alternatives

So, their arguments better be only one of `index` or `indices` for consistency.

**Comment by jbschlosser** [comment]
Note that for BC reasons it's unfortunately unlikely that we'll make these consistent.

**Comment by albanD** [comment]
I would agree with @jbschlosser that the BC-breaking pain is not worth the benefit.

One option in line with the array API (that only has take) is to make these positional-only (which most people do anyways) and then the name only for documentation purposes and can be changed without concerns.
